lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -53,201 +53,6 @@ def test_fig_close():
-        else:
-            import signal
-            os.kill(os.getpid(), signal.SIGINT)
-
-    target = getattr(plt, target_name)
-    timer = threading.Timer(1, interrupter)
-    fig = plt.figure()
-    fig.canvas.mpl_connect(
-        'draw_event',
-        lambda *args: print('DRAW', flush=True)
-    )
-    fig.canvas.mpl_connect(
-        'draw_event',
-        lambda *args: timer.start()
-    )
-    try:
-        target(**kwargs)
-    except KeyboardInterrupt:
-        print('SUCCESS', flush=True)
-
-
-@pytest.mark.backend('QtAgg', skip_on_importerror=True)
-@pytest.mark.parametrize("target, kwargs", [
-    ('show', {'block': True}),
-    ('pause', {'interval': 10})
-])
-def test_sigint(target, kwargs):
-    backend = plt.get_backend()
-    proc = WaitForStringPopen(
-        [sys.executable, "-c",
-         inspect.getsource(_test_sigint_impl) +
-         f"\n_test_sigint_impl({backend!r}, {target!r}, {kwargs!r})"])
-    try:
-        proc.wait_for('DRAW')
-        stdout, _ = proc.communicate(timeout=_test_timeout)
-    except Exception:
-        proc.kill()
-        stdout, _ = proc.communicate()
-        raise
-    print(stdout)
-    assert 'SUCCESS' in stdout
-
-
-def _test_other_signal_before_sigint_impl(backend, target_name, kwargs):
-    import signal
-    import matplotlib.pyplot as plt
-    plt.switch_backend(backend)
-    from matplotlib.backends.qt_compat import QtCore  # noqa
-
-    target = getattr(plt, target_name)
-
-    fig = plt.figure()
-    fig.canvas.mpl_connect('draw_event',
-                           lambda *args: print('DRAW', flush=True))
-
-    timer = fig.canvas.new_timer(interval=1)
-    timer.single_shot = True
-    timer.add_callback(print, 'SIGUSR1', flush=True)
-
-    def custom_signal_handler(signum, frame):
-        timer.start()
-    signal.signal(signal.SIGUSR1, custom_signal_handler)
-
-    try:
-        target(**kwargs)
-    except KeyboardInterrupt:
-        print('SUCCESS', flush=True)
-
-
-@pytest.mark.skipif(sys.platform == 'win32',
-                    reason='No other signal available to send on Windows')
-@pytest.mark.backend('QtAgg', skip_on_importerror=True)
-@pytest.mark.parametrize("target, kwargs", [
-    ('show', {'block': True}),
-    ('pause', {'interval': 10})
-])
-def test_other_signal_before_sigint(target, kwargs):
-    backend = plt.get_backend()
-    proc = WaitForStringPopen(
-        [sys.executable, "-c",
-         inspect.getsource(_test_other_signal_before_sigint_impl) +
-         "\n_test_other_signal_before_sigint_impl("
-            f"{backend!r}, {target!r}, {kwargs!r})"])
-    try:
-        proc.wait_for('DRAW')
-        os.kill(proc.pid, signal.SIGUSR1)
-        proc.wait_for('SIGUSR1')
-        os.kill(proc.pid, signal.SIGINT)
-        stdout, _ = proc.communicate(timeout=_test_timeout)
-    except Exception:
-        proc.kill()
-        stdout, _ = proc.communicate()
-        raise
-    print(stdout)
-    assert 'SUCCESS' in stdout
-    plt.figure()
```

Comment:
@QuLogic Do you remember why you added this plt.figure() line here (in #20907?)  AFAICT it is not needed for the test and also locks in the running interactive framework for the main process, so I'm going to remove it(?)

Reply:
I'm not seeing any reason for it to be there either right now, and don't recall anything specific.